public/plugin.tsx

```diff
@@ -459,55 +461,57 @@ export class ObservabilityPlugin
       return `${type.charAt(0).toUpperCase()}${type.slice(1)}`;
     };
 
-    // register all s3 datasources
-    const registerDataSources = () => {
-      try {
-        core.http.get(`${DATACONNECTIONS_BASE}`).then((s3DataSources) => {
-          s3DataSources.map((s3ds) => {
-            dataSourceService.registerDataSource(
-              dataSourceFactory.getDataSourceInstance(S3_DATA_SOURCE_TYPE, {
-                id: htmlIdGenerator(OBS_S3_DATA_SOURCE)(),
-                name: s3ds.name,
-                type: s3ds.connector.toLowerCase(),
-                metadata: {
-                  ...s3ds.properties,
-                  ui: {
-                    label: s3ds.name,
-                    typeLabel: getDataSourceTypeLabel(s3ds.connector.toLowerCase()),
-                    groupType: s3ds.connector.toLowerCase(),
-                    selector: {
-                      displayDatasetsAsSource: false,
+    // register all s3 datasources only if mds feature flag is disabled
+    if (!this.featureFlagStatus) {
```

Comment:
can we directly use `setupDeps.dataSource` here?

Reply:
we can't access the `setupDeps` from the `start` function, we can only access to the `startDeps`, However, the `startDeps.dataSource.dataSourceEnabled` is also not accessible, when the dataSource plugin is not registered.

Reply:
got it. this makes sense for now. Ideally with should stop the registration at setup itself. Adding if condition here: https://github.com/opensearch-project/dashboards-observability/blob/3ad2bf0448f6a705f449f42e62be8398f840ea37/public/plugin.tsx#L264C11-L264C31